packages/wagmi/package.json

```diff
@@ -45,12 +45,12 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "typescript": "^4.9.5",
-    "wagmi": "0.11.7"
+    "wagmi": "^0.12.6"
   },
   "peerDependencies": {
     "@lens-protocol/react-web": "workspace:*",
     "ethers": "^5.7.2",
-    "wagmi": ">=0.9.2 <=0.11.7"
+    "wagmi": "^0.12.6"
```

Comment:
Can we keep a range here so or is there a breaking change?

Reply:
I thought, as there are no issues with the latest one, what's the point of maintaining support for the previous versions? also, I can't guarantee that the previous versions still work with the latest SDK, as I didn't test it (back to 0.9).

Reply:
> what's the point of maintaining support for the previous versions?

In general is to avoid forcing all consumers of the Lens SDK to upgrade another dependency just because they upgraded `@lens-protocol/wagmi`.

Maybe not a big deal now that we are still in 0.x, but as we move towards 1.0 and beyond IMO we should not make the upgrade path an unnecessary burden on the SDK consumer.

Also as side note, the `^` for a dependency that is still in initial development phase (i.e. `0.x`) is to be avoided as there is no guarantees that a patch release will introduce a breaking change.

Regarding:

> I can't guarantee that the previous versions still work with the latest SDK, as I didn't test it

Do we need to? We know that `@lens-protocol/wagmi` did work with `0.9.2` at some point, it then worked with anything in between `0.9.2` and `0.11.7`. Finally we know it works with `0.12.6`. We also know we didn't have to make code changes to this package as we upgraded wagmi versions.

We also know that this package is an adapter for wagmi, so as such it provides an implementation that works with an interface defined in `@lens-protocol/react-web`. So, unless the interface is leaking (i.e. there are subtle coupling between `@lens-protocol/react-web` and `@lens-protocol/wagmi` internals) we can, with good confidence assume that it works without further ad-hoc testing. This is not, by any stretch of imagination a 100% assurances that the consumer will not find a bug with a wagmi version within that range, but IMO still OK given the scope of this project.

For sure having some tests setup in place that actually run some automated test with the exact versions of wagmi will be the way to get 100% confidence, but the ROI in such setup seems marginal.

Thoughts?

Reply:
For posterity, agreed to use fixed `0.12.6` wagmi dep ahead of v1 release.